src/main/java/edu/hawaii/its/api/controller/ErrorControllerAdvice.java

```diff
@@ -27,63 +31,165 @@ public class ErrorControllerAdvice {
+
+        ApiError.Builder errorBuilder = new ApiError.Builder()
+                .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                .message("Runtime Exception")
+                .debugMessage(exception.getMessage())
+                .timestamp(LocalDateTime.now());
+
+        ApiError apiError = errorBuilder.build();
+
+        return buildResponseEntity(apiError);
     }
 
     @ExceptionHandler({MessagingException.class, IOException.class})
-    public ResponseEntity<GroupingsHTTPException> handleMessagingException(Exception e) {
+    public ResponseEntity<Object> handleMessagingException(Exception e) {
         emailService.sendWithStack(e, "Messaging Exception");
-      return exceptionResponse("Mail service exception", e, 500);
+
+        ApiError.Builder errorBuilder = new ApiError.Builder()
+                .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                .message("Mail service exception")
+                .debugMessage(e.getMessage())
+                .timestamp(LocalDateTime.now());
+
+        ApiError apiError = errorBuilder.build();
+
+        return buildResponseEntity(apiError);
     }
 
     @ExceptionHandler(UnsupportedOperationException.class)
-    public ResponseEntity<GroupingsHTTPException> handleUnsupportedOperationException(
-        UnsupportedOperationException nie) {
-        emailService.sendWithStack(nie, "Unsupported Operation Exception");
-      return exceptionResponse("Method not implemented", nie, 501);
+    public ResponseEntity<Object> handleUnsupportedOperationException(
+        UnsupportedOperationException ex) {
+        emailService.sendWithStack(ex, "Unsupported Operation Exception");
+
+        ApiError.Builder errorBuilder = new ApiError.Builder()
+                .status(HttpStatus.NOT_IMPLEMENTED)
+                .message("Unsupported Operation Exception")
+                .debugMessage(ex.getMessage())
+                .timestamp(LocalDateTime.now());
+
+        ApiError apiError = errorBuilder.build();
+
+        return buildResponseEntity(apiError);
     }
 
-    private ResponseEntity<GroupingsHTTPException> exceptionResponse(String message, Throwable cause, int status) {
-        GroupingsHTTPException httpException = new GroupingsHTTPException(message, cause, status);
+    /* Custom Exception for UH GROUPINGS */
+
+    @ExceptionHandler(ExceptionForTesting.class)
+    protected ResponseEntity<Object> handleExceptionForTesting(ExceptionForTesting ex) {
+        emailService.sendWithStack(ex, "Exception For Testing Exception");
+
+        ApiError.Builder errorBuilder = new ApiError.Builder()
+                .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                .message("Validation failed")
+                .debugMessage("Check your service at endpoint /exception")
+                .timestamp(LocalDateTime.now());
+
+        if (ex.getSubErrors() != null) {
+            errorBuilder.addAllSubErrors(ex.getSubErrors());
+        }
```

Comment:
Could you put this null guard inside the errorBuilder.addAllSubErrors function? That we do not need to repeat this null guard for every exception we create. Thanks!